core/services/offchainreporting/peerstore.go

```diff
@@ -2,21 +2,153 @@ package offchainreporting
 
 import (
 	"context"
-	"database/sql"
+	"time"
 
-	sqlds "github.com/ipfs/go-ds-sql"
-	pgqueries "github.com/ipfs/go-ds-sql/postgres"
+	"github.com/jinzhu/gorm"
+	p2ppeer "github.com/libp2p/go-libp2p-core/peer"
 	p2ppeerstore "github.com/libp2p/go-libp2p-core/peerstore"
-	p2ppeerstoreds "github.com/libp2p/go-libp2p-peerstore/pstoreds"
+	"github.com/libp2p/go-libp2p-peerstore/pstoremem"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/postgres"
+	"github.com/smartcontractkit/chainlink/core/utils"
 )
 
-const tableName = "p2p_peerstore"
+type (
+	p2pPeer struct {
+		ID        p2ppeer.ID
+		Addr      ma.Multiaddr
+		JobID     int32
+		CreatedAt time.Time
+		UpdatedAt time.Time
+	}
 
-// NewPeerstore creates a new database-backed peerstore
-// NOTE: You can get sql.DB from store with store.DB.DB()
-func NewPeerstore(ctx context.Context, db *sql.DB) (p2ppeerstore.Peerstore, error) {
-	queries := pgqueries.NewQueries(tableName)
-	datastore := sqlds.NewDatastore(db, queries)
-	opts := p2ppeerstoreds.DefaultOpts()
-	return p2ppeerstoreds.NewPeerstore(ctx, datastore, opts)
+	Pstorewrapper struct {
```

Comment:
I think we can move it later, for now it is only used by offchainreporting so it might as well stay there to try to keep package explosion under control.